<a href="https://colab.research.google.com/github/tushwagh/Playstore-Data-analysis/blob/main/Play_Store_App_Review_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market. </b>

## <b> Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.</b>

## <b> Explore and analyze the data to discover key factors responsible for app engagement and success. </b>

In [1]:
# Loading the libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from datetime import datetime
import warnings
warnings.filterwarnings('ignore')
%matplotlib inline

sns.set_style("darkgrid")

In [2]:
# Mounting the drive where the dataset to work on is saved
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
# Defining the directory path where the data is stored
directory_path = "/content/drive/MyDrive/Playstore Dataset/"

In [4]:
# Loading the dataset as pandas data frame
apps_data = pd.read_csv(directory_path + "Play Store Data.csv")
user_reviews = pd.read_csv(directory_path + "User Reviews.csv")

## **Understanding the basics of dataset**

Let's first have a look at the structure of the dataset and understand how the data is organized.

In [5]:
# taking a glimpse of the dataset
apps_data.head()

App  ...   Android Ver
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...  4.0.3 and up
1                                Coloring book moana  ...  4.0.3 and up
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...  4.0.3 and up
3                              Sketch - Draw & Paint  ...    4.2 and up
4              Pixel Draw - Number Art Coloring Book  ...    4.4 and up

[5 rows x 13 columns]

In [6]:
# checking the shape of dataset
apps_data.shape

(10841, 13)

In [7]:
# looking at the columns
apps_data.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

In [8]:
# basic info
apps_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


### <b>*Insight of the data set*</b>

It’s important to know the different types of data/variables in the given dataset. These are the different types of data present in the dataset like float, object, string.

We can observe that, our dataset contains the data of about 10841 apps found on the play store.

Dataset has 13 columns which are the parameters of the apps. Let's look at each column -

*   **App** - Name of the app
*   **Category** - type of the app
*   **Rating** - rated by the users out of 5
*   **Reviews** - number of reviews given by users
*   **Size** - size of the app in mb
*   **Installs** - number of instalations of app
*   **Type** - free or paid
*   **Price** - price in $ of paid apps
*   **Content Rating** - rating for which users can use app
*   **Genres** - category or type of app
*   **Last Updated** - date on which app updated last time
*   **Current Ver** - current version of the app
*   **Android Ver** - android version which app supports



## **Analyzing and treating missing values**

Missing values are caused by incomplete data. It is important to handle missing values effectively, as they can lead to inaccurate inferences and conclusions.

**Let's start by analyzing variables with missing values.**

In [9]:
# checking the columns which have missing values
columns_with_missing_values = apps_data.columns[apps_data.isnull().any()]
apps_data[columns_with_missing_values].isnull().sum()

Rating            1474
Type                 1
Content Rating       1
Current Ver          8
Android Ver          3
dtype: int64

**Yep, we do have missing values.**

Looking at number of missing values above, we notice that the column *Rating* has a lot of missing values in it. The other columns do have missing values but they are less than 10.

Let's analyze the columns with missing values so we can figure out why the data is missing. This is the point at which we get into the part of data science. It can be a frustrating part of data science, especially if you're newer to the field and don't have a lot of experience. For dealing with missing values, we'll need to use our intution to figure out why the value is missing.

Let's start with column ***Type*** 

In [10]:
# Looking at missing values in column Type
apps_data[apps_data['Type'].isna()]

App  ...         Android Ver
9148  Command & Conquer: Rivals  ...  Varies with device

[1 rows x 13 columns]

There is only one missing value. These value is probably missing because it is not recorded, rather than because it doesn't exist. So, it would make sense for us to try and guess what it would be rather than just leaving is as NaN.

After cross-checking in the dataset the app's price is 0 it means the missing value is Free, So now we can fill the missing value with *Free*